### PR TITLE
Selective sync: hide child indicator for a childless folder

### PR DESCRIPTION
**Pull request by erikjv**
Fixes: #8935

### CODE REVIEW

File: src/gui/selectivesyncwidget.cpp [hunk 1]
```diff
@@ -174,7 +175,7 @@ void SelectiveSyncWidget::recursiveInsert(QTreeWidgetItem *parent, QStringList p
         }
 
         pathTrail.removeFirst();
-        recursiveInsert(item, pathTrail, path, size);
+        recursiveInsert(item, pathTrail, path, size, showChildIndicator);
```
Comment (TheOneRing) [inline src/gui/selectivesyncwidget.cpp:178]: Could you add a comment how showChildIndicator works in this recursion?
As it isn't changed, I'd expect it to define the indicator for the whole sub tree.

**Review by TheOneRing — CHANGES_REQUESTED** [review]
^

**Review by TheOneRing — APPROVED** [review]

File: src/gui/selectivesyncwidget.cpp [hunk 2]
```diff
@@ -138,14 +138,28 @@ static QTreeWidgetItem *findFirstChild(QTreeWidgetItem *parent, const QString &t
...
+ * @param path The full path
+ * @param size The size of the folder
+ * @param showChildIndicator False if it is known that a folder does not have any child items
+ *        (and should not show the expansion triangle), true otherwise.
+ *
+ * @note Note about the `showChildIndicator`: if set to false, this will only be applied to the very
+ *       last item of the path/pathTrail: all parent items need to be expandable in order to reach
+ *       that last folder. So all parents of that folder will have expansion triangles shown.
+ */
+void SelectiveSyncWidget::recursiveInsert(QTreeWidgetItem *parent, QStringList pathTrail, QString path, qint64 size, bool showChildIndicator)
```
Comment (erikjv) [inline src/gui/selectivesyncwidget.cpp:154]: Explain how the recursive algorithm works

Comment (erikjv) [reply src/gui/selectivesyncwidget.cpp:165]: done

**Review by TheOneRing — APPROVED** [review]

**Pull request merged**